staging/src/k8s.io/apiserver/pkg/registry/rest/update.go

```diff
@@ -110,6 +110,12 @@ func BeforeUpdate(strategy RESTUpdateStrategy, ctx context.Context, obj, old run
 		objectMeta.SetInitializers(nil)
 	}
 
+	// Ensure managedFields state is removed unless ServerSideApply is enabled
+	if !utilfeature.DefaultFeatureGate.Enabled(features.ServerSideApply) {
+		oldMeta.SetManagedFields(nil)
```

Comment:
We should have a comment about how the feature behaves when the flag is turned off and clients are using it / there are values in the DB - I know this is still under discussion though.